### PR TITLE
[Dataset] KarateClubDataset

### PR DESCRIPTION
**Pull request by HuXiangkun**
## Description
#1776

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/data/karate.py [hunk 1]
```diff
@@ -2,30 +2,81 @@
...
+    Nodes: 34
+    Edges: 156
+    Number of Classes: 2
+
+    Attributes
+    ----------
+    num_classes : int
+        Number of node classes
+
+    Returns
```
Comment (hetong007) [inline python/dgl/data/karate.py:37]: No need to have a return section

File: python/dgl/data/karate.py [hunk 2]
```diff
@@ -2,30 +2,81 @@
...
         assert idx == 0, "This dataset has only one graph"
-        return self.data[0]
+        return self._graph
 
     def __len__(self):
-        return len(self.data)
+        return 1
+
+
+KarateClub = KarateClubDataset
```
Comment (hetong007) [inline python/dgl/data/karate.py:82]: How about we also changes `KarateClub` to `KarateClubDataset` in our offical examples?

Comment (HuXiangkun) [reply python/dgl/data/karate.py:82]: We don't have examples using KarateClub.

File: python/dgl/data/karate.py [hunk 3]
```diff
@@ -2,30 +2,81 @@
...
-    "An Information Flow Model for Conflict and Fission in Small Groups" by Wayne W. Zachary. The
-    network became a popular example of community structure in networks after its use by Michelle
-    Girvan and Mark Newman in 2002.
+__all__ = ['KarateClubDataset', 'KarateClub']
 
-    This dataset has only one graph, with ndata 'label' means whether the node is belong to the "Mr. Hi" club.
-    """
 
+class KarateClubDataset(DGLDataset):
+    r"""
```
Comment (classicsong) [inline python/dgl/data/karate.py:15]: Please give a short description.

File: python/dgl/data/karate.py [hunk 4]
```diff
@@ -2,30 +2,81 @@
...
+
+    Statistics
+    ----------
+    Nodes: 34
+    Edges: 156
+    Number of Classes: 2
+
+    Attributes
+    ----------
+    num_classes : int
```
Comment (classicsong) [inline python/dgl/data/karate.py:34]: `data` is also a attributes
And also add deprecated description of `data`

**Pull request merged**